bundles/org.openhab.binding.plex/src/main/java/org/openhab/binding/plex/internal/handler/PlexServerHandler.java

```diff
@@ -0,0 +1,325 @@
+import org.openhab.binding.plex.internal.config.PlexServerConfiguration;
+import org.openhab.binding.plex.internal.dto.MediaContainer;
+import org.openhab.binding.plex.internal.dto.MediaContainer.MediaType;
+import org.openhab.core.i18n.ConfigurationException;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.util.ThingWebClientUtil;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link PlexServerHandler} is responsible for creating the
+ * Bridge Thing for a PLEX Server.
+ *
+ * @author Brian Homeyer - Initial contribution
+ * @author Aron Beurskens - Binding development
+ */
+@NonNullByDefault
+public class PlexServerHandler extends BaseBridgeHandler implements PlexUpdateListener {
+    private final Logger logger = LoggerFactory.getLogger(PlexServerHandler.class);
+
+    private final HttpClientFactory httpClientFactory;
+    private @Nullable HttpClient httpClient;
+
+    // Maintain mapping of handler and players
+    private final Map<String, PlexPlayerHandler> playerHandlers = new ConcurrentHashMap<>();
+
+    private PlexServerConfiguration config = new PlexServerConfiguration();
+    private PlexApiConnector plexAPIConnector;
+
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    private volatile boolean isRunning = false;
+
+    public PlexServerHandler(Bridge bridge, HttpClientFactory httpClientFactory) {
+        super(bridge);
+        this.httpClientFactory = httpClientFactory;
+        plexAPIConnector = new PlexApiConnector(scheduler, httpClientFactory.getCommonHttpClient());
+        logger.debug("Initializing server handler");
+    }
+
+    public PlexApiConnector getPlexAPIConnector() {
+        return plexAPIConnector;
+    }
+
+    /**
+     * Initialize the Bridge set the config paramaters for the PLEX Server and
+     * start the refresh Job.
+     */
+    @Override
+    public void initialize() {
+        final String httpClientName = ThingWebClientUtil.buildWebClientConsumerName(thing.getUID(), null);
+        try {
+            SslContextFactory.Server sslContextFactory = new SslContextFactory.Server();
+            sslContextFactory.setEndpointIdentificationAlgorithm(null);
+            httpClient = httpClientFactory.createHttpClient(httpClientName, sslContextFactory);
+            final HttpClient localHttpClient = this.httpClient;
+            if (localHttpClient != null) {
+                localHttpClient.start();
+                plexAPIConnector = new PlexApiConnector(scheduler, localHttpClient);
+            }
```

Comment:
This could be shortened:
```suggestion
            HttpClient localHttpClient = httpClient = httpClientFactory.createHttpClient(httpClientName,
                    sslContextFactory);
            localHttpClient.start();
            plexAPIConnector = new PlexApiConnector(scheduler, localHttpClient);
```

Reply:
It has been shortened.